routers/api/v1/api.go

```diff
@@ -1381,7 +1381,7 @@ func Routes() *web.Route {
 				m.Post("/{username}/{reponame}", admin.AdoptRepository)
 				m.Delete("/{username}/{reponame}", admin.DeleteUnadoptedRepository)
 			})
-			m.Group("/hooks", func() {
+			m.Group("/hooks/{configType:system|default}", func() {
```

Comment:
This is the core change. Almost everything else is fall-out from making this one line change.

Reply:
In https://github.com/go-gitea/gitea/pull/23142#discussion_r1185678106, lunny suggested `/api/v1/admin/hooks/default`, but also wanted to keep the old behaviour too. Respectfully, I think we must not. The old behaviour was inconsistent:

* GET /hooks -> return system hooks (because it [calls `GetSystemWebhooks`](https://github.com/go-gitea/gitea/pull/26418/files?diff=split&w=0#diff-22d0aa89fecea5f42f8b5b8bee1deff88580051e9723b162fa37810b1a9b708fL40))
* POST /hooks -> create a default hook (because [`isSystemWebhook` is missing therefore false](https://github.com/go-gitea/gitea/pull/26418/files?diff=split&w=0#diff-98b5cce3b814ddf612db812011c9192de3abcbbc6022dd39fa6d0fd02dc8ca25R188)).

We could chose to keep the `POST /hooks` == `POST /hooks/default/`, but then we would have to break `GET /hooks`; we could choose to keep `GET /hooks` == `GET /hooks/system`, but then we would have to break `POST /hooks`.

The confused behaviour was in the [original version of `/api/v1/admin/hooks`](https://github.com/go-gitea/gitea/pull/14537/), so I doubt anyone is using it in the wild -- I was probably the first person to try and immediately hit this bug. Anyone else who tried would have hit it too.